extensions/virtual-threads/runtime/src/main/java/io/quarkus/virtual/threads/ContextPreservingExecutorService.java

```diff
@@ -0,0 +1,105 @@
+package io.quarkus.virtual.threads;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.concurrent.Callable;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Future;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+import io.vertx.core.Vertx;
+import io.vertx.core.impl.ContextInternal;
+
+/**
+ * Delegating executor service implementation preserving the Vert.x context on {@link #execute(Runnable)}
+ */
+class ContextPreservingExecutorService implements ExecutorService {
```

Comment:
it looks like the mutiny binding one.
Do you plan to converge?

Reply:
We can move a variation of this to Mutiny bindings. But currently, it is not the same.

Reply:
CC @jponge

Reply:
Here's what we're doing https://github.com/smallrye/smallrye-mutiny-vertx-bindings/blob/main/vertx-mutiny-clients/vertx-mutiny-core/src/main/java/io/smallrye/mutiny/vertx/core/ContextAwareExecutorWrapper.java

If you think we can generalize/improve please let me know 👍